src/macro/macroApi.ml

```diff
@@ -1747,6 +1752,60 @@ let macro_api ccom get_api =
 			(get_api()).add_global_metadata (decode_string s1) (decode_string s2) (decode_bool b1,decode_bool b2,decode_bool b3) (get_api_call_pos());
 			vnull
 		);
+		"register_define_impl", vfun5 (fun s1 s2 a1 a2 a3 ->
+			let flags : define_parameter list = [] in
+
+			let platforms = decode_opt_array decode_string a1 in
+			let flags =
+				if (List.length platforms) = 0 then flags
+				else (Platforms (List.map (fun p -> (Globals.parse_platform p)) platforms)) :: flags
+			in
+
+			let params = decode_opt_array decode_string a2 in
+			let flags = List.append flags (List.map (fun p -> (HasParam p : define_parameter)) params) in
+
+			let links = decode_opt_array decode_string a3 in
+			let flags = List.append flags (List.map (fun l -> (Link l : define_parameter)) links) in
+
+			(get_api()).register_define (decode_string s1) ((decode_string s2), flags);
+			vnull
+		);
+		"register_metadata_impl", vfun6 (fun s1 s2 a1 a2 a3 a4 ->
+			let flags : meta_parameter list = [] in
+
+			let platforms = decode_opt_array decode_string a1 in
+			let flags =
+				if (List.length platforms) = 0 then flags
+				else (Platforms (List.map (fun p -> (Globals.parse_platform p)) platforms)) :: flags
+			in
+
+			let targets = decode_opt_array decode_string a2 in
+			let flags =
+				if (List.length targets) = 0 then flags
+				else (UsedOn (List.map (function
```

Comment:
I don't really like how separate it is from actual definition in `prebuild.ml`
Any suggestion how to improve? Maybe generate this as a `parse_target` or something in metaList.ml via prebuild.ml?

Reply:
"target" isn't a great name here, but yeah, generating a parse function seems sensible.